crates/driver/src/infra/solver/dto/auction.rs

```diff
@@ -161,7 +162,16 @@ impl Auction {
                         ),
                         app_data: AppDataHash(order.app_data.hash().0.into()),
                         flashloan_hint: flashloans_enabled
-                            .then(|| order.app_data.flashloan().cloned().map(Into::into))
+                            .then(|| {
+                                order.app_data.flashloan().map(|flashloan| FlashloanHint {
```

Comment:
I wonder if filling in these default values makes more sense in the component that fetches the app data in the first place. 🤔 
No strong opinion, though since I can see why one would want to have access to the pure unaltered appdata.

Reply:
True. For now, flashloan hints in their final shape are not needed/used in the driver domain. But we can move this logic out to driver domain when then need comes.